api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -214,13 +216,24 @@ func (c *imageBuilder) imageRefExists(imageName, imageTag string) (bool, error)
 	tags, err := remote.List(repo, remote.WithAuthFromKeychain(keychain))
 	if err != nil {
 		if terr, ok := err.(*transport.Error); ok {
+			log.Errorf("transport error on listing tags for %s: status code: %d, error: %s", imageName, terr.StatusCode, terr.Error())
+
 			// If image not found, it's not exist yet
 			if terr.StatusCode == http.StatusNotFound {
 				return false, nil
 			}
-		} else {
-			return false, errors.Wrapf(err, "error getting image tags for %s", repo)
+
+			// If unauthorized, it's more likely due to expired token. So, let's try again
+			if terr.StatusCode == http.StatusUnauthorized {
+				if retryCounter < maxCheckImageRetry {
+					log.Infof("%d retry listing tags for %s", retryCounter+1, imageName)
+					return c.imageRefExists(imageName, imageTag, retryCounter+1)
```

Comment:
can we add delay between retry, let say 1s?

Reply:
Added. I was thinking about it before, but forget to implement it. Thanks for the reminder!